src/main/views/webContentEvents.ts

```diff
@@ -177,7 +177,11 @@ export class WebContentsEventManager {
 
             // TODO: move popups to its own and have more than one.
             if (urlUtils.isPluginUrl(serverURL, parsedURL) || urlUtils.isManagedResource(serverURL, parsedURL)) {
-                if (!this.popupWindow) {
+                if (this.popupWindow) {
+                    this.popupWindow.once('ready-to-show', () => {
+                        this.popupWindow!.show();
```

Comment:
It should not need the bang

Reply:
Whoops 😓

Reply:
It does because when the callback is called it can't guarantee that the object exists (but it obviously does)

Reply:
What about optional chaining instead? `this.popupWindow?.show()`

Reply:
Since it's written as `this.popupWindow`, I think it might be possible for `this.popupWindow` to change between the the call to `once` and when this callback is actually used. If you were to capture the value like
```ts
const popupWindow = this.popupWindow;
popupWindow.once('ready-to-show', () => popupWindow.show());
```
I think it would be fine without the `!`